Fix issues in activation_memory_budget for float8

y-sq opened the pull request:
Summary:
When using activation_memory_budget for float8 training, two issues were noticed:

- When `aggressive_options` (https://fburl.com/code/m1yoskxw) is called , all fp8 gemms (the scaled_mm op) are saved for recomputation.
- After adding "scaled_mm" in the `compute_intensive_ops`, we got the next error from `estimate_runtime`: `mat2 must be col_major` from `meta_scaled_mm`.
To fix it, modified `materialize_arg` to also include the stride of the original tensor.

Test Plan: Run float8 training with `activation_memory_budget`.

Differential Revision: D60777297

Chillee reviewed (approved):
LGTM.

The stride change is good - thanks!

One thing we should change is we should add a tag for "compute intensive ops".

y-sq committed:
Fix issues in activation_memory_budget for float8 (#132687)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/132687

When using activation_memory_budget for float8 training, two issues were noticed:

- When `aggressive_options` (https://fburl.com/code/m1yoskxw) is called , all fp8 gemms (the scaled_mm op) are saved for recomputation.
- After adding "scaled_mm" in the `compute_intensive_ops`, we got the next error from `estimate_runtime`: `mat2 must be col_major` from `meta_scaled_mm`.
To fix it, modified `materialize_arg` to also include the stride of the original tensor.

Test Plan: Run float8 training with `activation_memory_budget`.

Reviewed By: Chillee

Differential Revision: D60777297

y-sq commented:
@pytorchbot merge

y-sq commented:
@pytorchbot label "topic: not user facing"

y-sq commented:
@pytorchbot merge
